### PR TITLE
feat(semseg): allow model customization

### PR DESCRIPTION
**Pull request by borisdayma**
Replacing #1338 for rebasing

# Before submitting

- [X] Was this discussed/approved via a Github issue? (no need for typos and docs improvements)
- [X] Did you read the [contributor guideline](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CONTRIBUTING.md), Pull Request section?
- [X] Did you make sure to update the docs?   
- [ ] Did you write any new necessary tests?  
- [X] If you made a notable change (that affects users), did you update the [CHANGELOG](https://github.com/PyTorchLightning/pytorch-lightning/blob/master/.github/CHANGELOG.md)?

## What does this PR do?
Allow more customization to the semantic segmentation example:

* test set has been replace with a validation set to measure model performance
* both training and validation losses are logged
* unet allows for more customization: number of layers and number of features
* masks have been updated to be compatible with the most recent version of the dataset which contains more labels (additional labels are just ignored)
* script allows for other arguments: distributed backend, 16 bit, number of layers and features for unet as well as type of upsampling, gradient accumulation, number of epochs, logging on W&B

## PR review    
Anyone in the community is free to review the PR once the tests have passed.     
If we didn't discuss your PR in Github issues there's a high chance it will not be merged.

## Did you have fun?
Make sure you had fun coding 🙃

### CODE REVIEW

**Comment by Borda** [comment]
@borisdayma what is the diff to #1338 while it has the same name?

**Comment by borisdayma** [comment]
From [this comment](https://github.com/PyTorchLightning/pytorch-lightning/pull/1338#issuecomment-608610028) I understood I had to rebase and make a new PR.

**Comment by Borda** [comment]
> From [this comment](https://github.com/PyTorchLightning/pytorch-lightning/pull/1338#issuecomment-608610028) I understood I had to rebase and make a new PR.

ok, it seems as a misunderstanding, rebase was sufficient, no need for creating new PR lol

**Comment by Borda** [comment]
changelog need to be rebased on new release #1419

**Comment by borisdayma** [comment]
Sorry but I'm not sure of the process of rebasing a PR. Would github allow me to rewrite previous history? I'm typically used to merge commits.

**Comment by Borda** [comment]
yes, it does, and just when mentioned PR is merged I can do it for you so you do not need to do anything... :]

**Comment by borisdayma** [comment]
Thanks a lot for the changes!
